modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionServiceImpl.java

```diff
@@ -426,8 +426,9 @@ public BaseModelSearchResult<CPDefinition> searchCPDefinitions(
 		return cpDefinitionLocalService.searchCPDefinitions(
 			companyId,
 			TransformUtil.transformToLongArray(
-				_commerceCatalogService.getCommerceCatalogs(
-					companyId, QueryUtil.ALL_POS, QueryUtil.ALL_POS),
+				_commerceCatalogService.search(
```

Comment:
Can you explain why using the index instead of the DB call here? thanks

Reply:
Hey @luca-pellizzon I changed it to the index call because it does the filtering based on our filtering logic. That specific DB call is using filterFindByCompanyId. It also does some permission checking but was not going through our catalog permission checking logic.